shaptools/hana.py

```diff
@@ -447,6 +447,193 @@ def get_sr_status(self):
+        reconfig_option = ' WITH RECONFIGURE' if reconfig else ''
+
+        cmd = ('{hdbsql_cmd} -d {db} '
+               '\"ALTER SYSTEM ALTER CONFIGURATION(\'{file_name}\', \'{layer}\'{layer_name}) SET'
+               '{parameter_list_str}{reconfig};\"'.format(
+                   hdbsql_cmd=hdbsql_cmd, db=database, file_name=file_name, layer=layer,
+                   layer_name=layer_name_str, parameter_list_str=parameter_list_str,
+                   reconfig=reconfig_option))
+
+        # TODO: return the HANA SQL Statement error if sql fails
+        self._run_hana_command(cmd)
+
+    def unset_ini_parameter(
+            self, parameter_list,
+            database, file_name, layer,
+            **kwargs):
+        """
+        unset HANA configuration parameters in ini file
+
+        sql syntax:
+        ALTER SYSTEM ALTER CONFIGURATION (<filename>, <layer>[, <layer_name>])
+        UNSET (<section_name>,<parameter_name>);
+
+        Args:
+            parameter_list(list)
+                section_name (str): section name of parameter in ini file
+                parameter_name (str): name of the parameter to be modified
+            database (str): Database name
+            file_name (str): ini configuration file name
+            layer (str): target layer for the configuration change 'SYSTEM', 'HOST' or 'DATABASE'
+            layer_name (str, optional): target either a tenant name or a target host name
+            reconfig (str, optional): if apply changes to running HANA instance
+            key_name (str, optional): Keystore to connect to sap hana db
+            user_name (str, optional): User to connect to sap hana db
+            user_password (str, optiona): Password to connecto to sap hana db
+        """
+        parameter_list_str = ', '.join(str(params) for params in parameter_list)
+
+        key_name = kwargs.get('key_name', None)
+        user_name = kwargs.get('user_name', None)
+        user_password = kwargs.get('user_password', None)
+        hdbsql_cmd = self._hdbsql_connect(
+            key_name=key_name, user_name=user_name, user_password=user_password)
+
+        layer_name = kwargs.get('layer_name', None)
+        if layer in ('HOST', 'DATABASE') and layer_name is not None:
+            layer_name_str = ", \'" + layer_name + "\'"
+        else:
+            layer_name_str = ''
+
+        reconfig = kwargs.get('reconfig', False)
+        reconfig_option = ' WITH RECONFIGURE' if reconfig else ''
+
+        cmd = ('{hdbsql_cmd} -d {db} '
+               '\"ALTER SYSTEM ALTER CONFIGURATION(\'{file_name}\', \'{layer}\'{layer_name}) UNSET'
+               '{parameter_list_str}{reconfig};\"'.format(
+                   hdbsql_cmd=hdbsql_cmd, db=database, file_name=file_name, layer=layer,
+                   layer_name=layer_name_str, parameter_list_str=parameter_list_str,
+                   reconfig=reconfig_option))
+
+        self._run_hana_command(cmd)
+
+    def update_memory_resources(
```

Comment:
I don't understand the usage of this method. It's doing exactly the same than set_ini_parameter.
What's the purpose?

If we want to change the memorymanager and system_replication values, shouldn't be hardcoded here?
Rethink what do you want to do here

Reply:
Hi @arbulu89 yes, this is what I wanted to actually discuss, as in my discussion above(from last week).

This method does not make sense here anymore.

I created this original method (reduce_memory_resources) basically to reduce hana resources taking only Global allocation limit as variable argument, and putting that in a hardcoded SQL string to reduce memory resources. Because 'Global allocation limit' is the only variable parameter for reducing resources, other one being preload_column_tables:False

But right now I moved that logic of hardcoding the parameter names for SQL to salt-saphana state. Let me know where you think it makes more sense, to define the parameter names(dict).

parameters = {('system_replication', 'preload_column_tables'): <value>, (\
                       'memorymanager', 'global_allocation_limit') : <global_allocation_limit_value>}

Reply:
Yes, it makes more sense, or these functions should host the logic to calculate the parameters values and call the other methods to effectively change the value.

Reply:
Hi @diegoakechi I plan to add the method, where I calculate/set the optimal memory allocation, and it can be hardcoded or better it passes those values to set_ini_parameter methods.
So I will remove/update these method with that functionality.

Reply:
@Simranpal Ok, In that case, I prefer you remove it from this PR to clean up the code, and create a new PR with these methods in the future.